### PR TITLE
first attempt on validation rules

### LINKED ISSUE #134: Finish Validation Rules
https://drafts.oauth.net/draft-ietf-oauth-status-list/draft-ietf-oauth-status-list.html#name-validation-rules

### PR DESCRIPTION
**Pull request by c2bo**
Closes #134 
Rendered Version: https://drafts.oauth.net/draft-ietf-oauth-status-list/c2bo/validation-rules/draft-ietf-oauth-status-list.html

### CODE REVIEW

File: draft-ietf-oauth-status-list.md [hunk 1]
```diff
@@ -447,7 +446,27 @@ The HTTP response SHOULD use gzip Content-Encoding as defined in {{RFC9110}}.
...
+    1. Validate the Status List Token by following the rules defined in section 7.2 of {{RFC7519}} for JWTs and section 7.2 of {{RFC8392}} for CWTs.
+    2. Check for the existence of the required claims as defined in [](#status-list-token-jwt) and [](#status-list-token-cwt)
+4. All existing claims in the Status List Token MUST be checked according to the rules in [](#status-list-token-jwt) and [](#status-list-token-cwt)
+    1. The subject claim (`sub` or `2`) of the Status List Token MUST be equal to the `uri` claim in the `status_list` object of the Referenced Token
+    2. If the Relying Party has custom policies regarding the freshness of the Status List Token, it SHOULD check the issued at claim (`iat` or `6`)
+    3. If expiration time is defined (`exp` or `4`), it MUST be checked if the Status List Token is expired
+    4. If the Referenced Token contains an issuer claim, the Status List Token MUST contain the same issuer claim (`iss` or `1`)
+    5. If the Relying Party is using a system for caching the Status List Token, it SHOULD check the `ttl` claim of the Status List Token and retrieve a fresh copy if (issued at + ttl < current time)
```
Comment (paulbastian) [inline draft-ietf-oauth-status-list.md:464]: What if ttl is not present?

File: draft-ietf-oauth-status-list.md [hunk 2]
```diff
@@ -447,7 +446,27 @@ The HTTP response SHOULD use gzip Content-Encoding as defined in {{RFC9110}}.
...
+5. Decompress the Status List with a decompressor that is compatible with DEFLATE {{RFC1951}} and ZLIB {{RFC1950}}
+6. Retrieve the status value of the index specified in the Referenced Token as described in [](#status-list)
```
Comment (paulbastian) [inline draft-ietf-oauth-status-list.md:466]: Check index value is within size of the byte array?

**Review by paulbastian — APPROVED** [review]

**Comment by paulbastian** [comment]
I think there is room for improvement but I'd rather start with something to get feedback

File: draft-ietf-oauth-status-list.md (hunk 1 above)
Comment (tplooker) [reply draft-ietf-oauth-status-list.md:464]:
```suggestion
    5. If the Relying Party is using a system for caching the Status List Token, it SHOULD check the `ttl` claim of the Status List Token and retrieve a fresh copy if (time status was resolved + ttl < current time)
```

Its not IAT you add the TTL too in order to determine whether to resolve an update its the time of resolving the status list.

Comment (c2bo) [reply draft-ietf-oauth-status-list.md:464]:
```suggestion
    5. If the `ttl` claim is present and the Relying Party is using a system for caching the Status List Token, it SHOULD check the `ttl` claim of the Status List Token and retrieve a fresh copy if (time status was resolved + ttl < current time)
```

File: draft-ietf-oauth-status-list.md (hunk 2 above)
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:466]:
```suggestion
6. Retrieve the status value of the index specified in the Referenced Token as described in [](#status-list). Fail if the provided index is out of bound of the status list
```

File: draft-ietf-oauth-status-list.md [hunk 3]
```diff
@@ -449,7 +448,27 @@ The HTTP response SHOULD use gzip Content-Encoding as defined in {{RFC9110}}.
 
 ## Validation Rules
 
-TBD
+Upon receiving a Referenced Token, a Relying Party MUST perform the validation of the Referenced Token - e.g., checking for expected attributes, valid signature, expiration time.
```
Comment (tplooker) [inline draft-ietf-oauth-status-list.md:451]:
```suggestion
Upon receiving a Referenced Token, a Relying Party MUST first perform the validation of the Referenced Token - e.g., checking for expected attributes, valid signature, expiration time.
```

File: draft-ietf-oauth-status-list.md [hunk 4]
```diff
@@ -449,7 +448,27 @@ The HTTP response SHOULD use gzip Content-Encoding as defined in {{RFC9110}}.
 
 ## Validation Rules
 
-TBD
+Upon receiving a Referenced Token, a Relying Party MUST first perform the validation of the Referenced Token - e.g., checking for expected attributes, valid signature, expiration time.
+As this is out of scope of this document, this validation is not be described here, but is expected to be done according to the format of the Referenced Token.
+
+If this validation was not successful, the Referenced Token MUST be rejected. If the validation was successful, the Relying Party MUST perform the following validation for the Status List:
```
Comment (tplooker) [inline draft-ietf-oauth-status-list.md:454]:
```suggestion
If this validation was not successful, the Referenced Token MUST be rejected. If the validation was successful, the Relying Party MUST perform the following validation steps to evaluate the status of the reference token:
```

File: draft-ietf-oauth-status-list.md [hunk 5]
```diff
@@ -449,7 +448,27 @@ The HTTP response SHOULD use gzip Content-Encoding as defined in {{RFC9110}}.
...
 ## Validation Rules
 
-TBD
+Upon receiving a Referenced Token, a Relying Party MUST first perform the validation of the Referenced Token - e.g., checking for expected attributes, valid signature, expiration time.
+As this is out of scope of this document, this validation is not be described here, but is expected to be done according to the format of the Referenced Token.
+
+If this validation was not successful, the Referenced Token MUST be rejected. If the validation was successful, the Relying Party MUST perform the following validation steps to evaluate the status of the reference token:
+
+1. Check for the existence of a `status` claim, check for the existence of a `status_list` claim within the `status` claim and validate that the content of `status_list` adheres to the rules defined in [](#referenced-token-jwt) for JWTs and [](#referenced-token-cwt) for CWTs. This step can be overruled if defined within the Referenced Token Format natively
+2. Resolve the Status List from the provided URI or make sure it was provided in other ways (e.g., by caching)
```
Comment (tplooker) [inline draft-ietf-oauth-status-list.md:457]:
```suggestion
2. Resolve the Status List from the provided URI
```

**Review by tplooker — APPROVED** [review]

**Pull request merged**